ext/hathach/tusb_port.cpp.in

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2020, Erik Henriksson
+ * Copyright (c) 2020, Niklas Hauser
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include <modm/platform/device.hpp>
+#include <modm/architecture/interface/interrupt.hpp>
+#include "tusb.h"
+
+%% for port, vectors in port_vectors.items() | sort
+	%% for vector in vectors | sort
+MODM_ISR({{vector}})
+{
+	tud_int_handler({{port}});
```

Comment:
I don't think this is completely correct, at least on SAM there is only one ISR handler but it can have multiple UART ports assigned to it

Reply:
Ok, I'll check. I basically just ack'ed myself through the `hw/bsp/` folder…